theano/scalar/tests/test_basic.py

```diff
@@ -66,7 +67,27 @@ def tes_mod(self):
             self.assertTrue(fn(a, b) == a%b, (a,))
 
 
+def has_f16(comp):
+    if any(i.type == float16 for i in comp.fgraph.inputs):
+        return True
+    for n in comp.fgraph.apply_nodes:
+        if any(o.type == float16 for o in n.outputs):
```

Comment:
this don't assert constant don't have float16.

What only doing a loop on comp.fgraph.variables and remove the line 71-72 at the same time.